packages/ckeditor5-engine/tests/view/observer/selectionobserver.js

```diff
@@ -162,6 +162,46 @@ describe( 'SelectionObserver', () => {
 		changeDomSelection();
 	} );
 
+	it( 'should not pass non-valid object on Firefox', done => {
+		testUtils.sinon.stub( env, 'isGecko' ).value( true );
+		testUtils.sinon.stub( Object.prototype, 'toString' ).value( () => null );
+
+		const spy = sinon.spy( selectionObserver.mutationObserver, 'flush' );
+
+		changeDomSelection();
+		setTimeout( () => {
+			sinon.assert.notCalled( spy );
+			done();
+		}, 100 );
+	} );
+
+	it( 'should detect restricted object on Firefox', done => {
+		testUtils.sinon.stub( env, 'isGecko' ).value( true );
+		testUtils.sinon.stub( Object.prototype, 'toString' ).throws(
+			new Error( 'Permission denied to access property Symbol.toStringTag' )
+		);
+
+		const spy = sinon.spy( selectionObserver.mutationObserver, 'flush' );
+
+		changeDomSelection();
+		setTimeout( () => {
+			sinon.assert.notCalled( spy );
+			done();
+		}, 100 );
+	} );
+
+	it( 'should pass valid object on Firefox', done => {
+		testUtils.sinon.stub( env, 'isGecko' ).value( true );
+
+		const spy = sinon.spy( selectionObserver.mutationObserver, 'flush' );
+
+		changeDomSelection();
+		setTimeout( () => {
+			sinon.assert.calledOnce( spy );
+			done();
+		}, 100 );
+	} );
```

Comment:
Let's wrap these into a `describe( 'Restricted objects handling in Gecko', () => {} )` group because... they are a group of tests for a very specific use-case 🙂